### PR TITLE
Scanbuild and other warnings fixed

### LINKED ISSUE #4290: -Wmaybe-uninitialized warning still around on RC1
FYI: The -Wmaybe-uninitialized warning has been around for a long time. I thought it would have been removed for the RC release. BTW, I am compiling a RELEASE version on an up to date Lubuntu 16.04.3 LTS Desktop. Thanks for the RC release, it looks good so far.
..........
[ 69%] Building C object libfreerdp/CMakeFiles/freerdp.dir/primitives/prim_sign.c.o
[ 69%] Building C object libfreerdp/CMakeFiles/freerdp.dir/primitives/prim_YUV.c.o
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/primitives/prim_YUV.c: In function ‘general_RGBToAVC444YUV’:
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/primitives/prim_YUV.c:1027:8: warning: ‘b5’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     *b5++ = V2;
        ^
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/primitives/prim_YUV.c:968:28: note: ‘b5’ was declared here
  BYTE* b1, *b2, *b3, *b4, *b5, *b6, *b7;
                            ^
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/primitives/prim_YUV.c:1025:8: warning: ‘b4’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     *b4++ = U2;
        ^
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/primitives/prim_YUV.c:968:23: note: ‘b4’ was declared here
  BYTE* b1, *b2, *b3, *b4, *b5, *b6, *b7;
                       ^
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/primitives/prim_YUV.c:893:8: warning: ‘b5’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     *b5++ = V2;
        ^
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/primitives/prim_YUV.c:833:28: note: ‘b5’ was declared here
  BYTE* b1, *b2, *b3, *b4, *b5, *b6, *b7;
                            ^
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/primitives/prim_YUV.c:891:8: warning: ‘b4’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     *b4++ = U2;
        ^
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/primitives/prim_YUV.c:833:23: note: ‘b4’ was declared here
  BYTE* b1, *b2, *b3, *b4, *b5, *b6, *b7;
                       ^
[ 69%] Building C object libfreerdp/CMakeFiles/freerdp.dir/primitives/prim_YCoCg.c.o
[ 70%] Building C object libfreerdp/CMakeFiles/freerdp.dir/primitives/primitives.c.o
......

And then there is the missing braces warning which I also thought was gone:

.......
[ 56%] Building C object libfreerdp/CMakeFiles/freerdp.dir/core/mcs.c.o
[ 57%] Building C object libfreerdp/CMakeFiles/freerdp.dir/core/nla.c.o
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/core/nla.c: In function ‘nla_encrypt_public_key_echo’:
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/core/nla.c:1013:2: warning: missing braces around initializer [-Wmissing-braces]
  SecBuffer Buffers[2] = { 0 };
  ^
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/core/nla.c:1013:2: warning: (near initialization for ‘Buffers[0]’) [-Wmissing-braces]
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/core/nla.c: In function ‘nla_decrypt_public_key_echo’:
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/core/nla.c:1071:2: warning: missing braces around initializer [-Wmissing-braces]
  SecBuffer Buffers[2] = { 0 };
  ^
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/core/nla.c:1071:2: warning: (near initialization for ‘Buffers[0]’) [-Wmissing-braces]
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/core/nla.c: In function ‘nla_encrypt_ts_credentials’:
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/core/nla.c:1417:2: warning: missing braces around initializer [-Wmissing-braces]
  SecBuffer Buffers[2] = { 0 };
  ^
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/core/nla.c:1417:2: warning: (near initialization for ‘Buffers[0]’) [-Wmissing-braces]
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/core/nla.c: In function ‘nla_decrypt_ts_credentials’:
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/core/nla.c:1471:2: warning: missing braces around initializer [-Wmissing-braces]
  SecBuffer Buffers[2] = { 0 };
  ^
/usr/local/src/FreeRDP-2.0.0-rc1/libfreerdp/core/nla.c:1471:2: warning: (near initialization for ‘Buffers[0]’) [-Wmissing-braces]
[ 57%] Building C object libfreerdp/CMakeFiles/freerdp.dir/core/nego.c.o
[ 57%] Building C object libfreerdp/CMakeFiles/freerdp.dir/core/info.c.o
.......

### PR DESCRIPTION
**Pull request by akallabeth**
Fix some compiler and scanbuild warnings

Fixes #4290

### CODE REVIEW

File: winpr/libwinpr/sspi/Schannel/schannel.c [hunk 1]
```diff
@@ -269,36 +247,30 @@ SECURITY_STATUS SEC_ENTRY schannel_AcceptSecurityContext(PCredHandle phCredentia
...
-
 		context->server = TRUE;
-
 		sspi_SecureHandleSetLowerPointer(phNewContext, context);
 		sspi_SecureHandleSetUpperPointer(phNewContext, (void*) SCHANNEL_PACKAGE_NAME);
-
 		schannel_openssl_server_init(context->openssl);
 	}
 
 	status = schannel_openssl_server_process_tokens(context->openssl, pInput, pOutput);
```
Comment (bmiklautz) [inline winpr/libwinpr/sspi/Schannel/schannel.c]: If status is only used for return. Possibly we should do a:
```c
return schannel_openssl_server_process_tokens(context->openssl, pInput, pOutput);
```
And drop status completely

File: libfreerdp/primitives/prim_YUV.c [hunk 2]
```diff
@@ -830,7 +830,13 @@ static INLINE pstatus_t general_RGBToAVC444YUV_BGRX(
 	 */
 	UINT32 x, y, n, numRows, numCols;
 	BOOL evenRow = TRUE;
-	BYTE* b1, *b2, *b3, *b4, *b5, *b6, *b7;
+	BYTE* b1 = NULL;
```
[the remark this answers is not in the document]
Comment (akallabeth) [reply libfreerdp/primitives/prim_YUV.c]: @mfleisz Did confuse some compilers and spilled warnings

**Comment by akallabeth** [comment]
@bmiklautz @mfleisz @hardening anyone want to merge?

File: channels/geometry/client/geometry_main.c [hunk 3]
```diff
@@ -170,33 +172,34 @@ static UINT geometry_recv_pdu(GEOMETRY_CHANNEL_CALLBACK* callback, wStream* s)
...
+
 	if (cbGeometryBuffer)
 	{
 		ret = geometry_read_RGNDATA(s, cbGeometryBuffer, &packet.geometry);
+
 		if (ret != CHANNEL_RC_OK)
 			return ret;
 	}
+	else
+		ret = CHANNEL_RC_OK;
```
Comment (hardening) [inline channels/geometry/client/geometry_main.c]: good catch

**Pull request merged**